File: 2018/12-Dec/12-22-multiple-inheritance/php/BalsamFir.php
<?php
/**
 * Create a class for a balsam fir tree, which is both a christmas tree and an evergreen tree
 * User: Andrew Jarombek
 * Date: 12/17/2018
 */

class BalsamFir implements Tree
{
    // Add traits to the class
    use EvergreenTree;
    use ChristmasTree;

    private $height_feet;
    private $height_inches;

    public function __construct($height)
    {
        $this->height_feet = $height["feet"];
        $this->height_inches = $height["inches"];
    }

    public function height()
    {
        return sprintf("%d' %d\"", $this->height_feet, $this->height_inches);
    }
}

File: 2018/12-Dec/12-22-multiple-inheritance/csharp/BalsamFir.cs
/**
 * Concrete class for a Balsam Fir which implements Tree and is a composition of ChristmasTree and EvergreenTree.
 * Author: Andrew Jarombek
 * Date: 1/14/2019
 */

namespace Multiple_Inheritance
{
    public sealed class BalsamFir : Tree
    {
        // private immutable fields used for composition
        private readonly ChristmasTree _christmasTree = new ChristmasTree();
        private readonly EvergreenTree _evergreenTree = new EvergreenTree();
        
        // BalsamFir properties
        private int Feet { get; }
        private int Inches { get; }

        /// <summary>
        /// Construct a balsam fir tree
        /// </summary>
        /// <param name="feet">The height of the tree in feet (floored)</param>
        /// <param name="inches">The height of the tree in inches</param>
        public BalsamFir(int feet, int inches)
        {
            Feet = feet;
            Inches = inches;
        }

        /// <inheritdoc cref="Tree.Type"/>
        public string Type() => _christmasTree.Type();

        /// <inheritdoc cref="Tree.Height"/>
        public int Height() => Feet * 12 + Inches;

        // Whether or not the leaves persist in winter
        public bool LeafPersistence => _evergreenTree.LeafPersistence;
    }
}